### PR TITLE
Clean up library dependencies

### PR DESCRIPTION
**Pull request by JBirdVegas**
Biggest change is to add support-annotations and prefer them
over jetbrains annotations which we still use for `@Contract`

Get jetbrains annotations, android support annotations and
android support library v4 from repositories.

Remove now unused jars from libs/

Seems we are using analytics v2 which is `@Deprecated`.  Once
the change is made to use v3 then we can get analytic jar
from repositories as well.

### CODE REVIEW

**Comment by JBirdVegas** [comment]
@p4r4n01d Here is a reference for the android support annotations. http://tools.android.com/tech-docs/support-annotations

Should give us more control in specifying inputs if we want.

File: app/src/main/java/com/jbirdvegas/mgerrit/views/DiffTextView.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ private void colorizeDiffs(String[] split, SpannableString spannableString,
                         charCounter + startWhitespace - 1, end, 0);
             }
 // test line with trailing whitespaces ->
-// Here are 3 tabs ->	-	-	<- this line ends with four whitespaces ->    
+// Here are 3 tabs ->	-	-	<- this line ends with four whitespaces ->
```
Comment (p4r4n01d) [inline app/src/main/java/com/jbirdvegas/mgerrit/views/DiffTextView.java]: May as well delete this line completely - especially if the comment is no longer relevant.

**Comment by p4r4n01d** [comment]
Do make sure that the rotating issue is not present. Last time I tried to update support libraries, rotating the device made the app crash. That was crashing on both phones and tablets (nexus 4 and rk30).

Side note: you seemed to inadvertently mention other github users in your original comment.

File: app/src/main/java/com/jbirdvegas/mgerrit/views/DiffTextView.java (hunk 1 above)
Comment (JBirdVegas) [inline app/src/main/java/com/jbirdvegas/mgerrit/views/DiffTextView.java]: Yup will do. It was only used when the diff parser was being developed

**Comment by JBirdVegas** [comment]
Oops updated comment. I'll test for a rotation crash later today.  If its still there we can drill down or revert the updated version

**Comment by JBirdVegas** [comment]
This is redundant now... closing

**Pull request closed without merging**